tidoop-mr-lib-api/src/server.js

```diff
@@ -23,36 +23,134 @@
  * Author: frb
  */
 
-// requires
+// module dependencies
 var Hapi = require('hapi');
-var config = require('../conf/tidoop-mr-lib-api.json');
-var pjson = require('../package.json');
+var run = require('./cmd_runner.js').run;
+var version = require('../package.json').version;
+var port = require('../conf/tidoop-mr-lib-api.json').port;
+var tidoopMRLibPath = require('../conf/tidoop-mr-lib-api.json').tidoopMRLibPath;
+var tidoopMysql = require('./mysql_driver.js');
+var serverUtils = require('./server_utils.js');
 
 // create a server with a host and port
 var server = new Hapi.Server();
 
 server.connection({ 
     host: 'localhost',
-    port: config.port
+    port: port
 });
 
-// add the route
+// create a connection to the MySQL server
+tidoopMysql.connect();
+
+// add routes
 server.route({
     method: 'GET',
-    path:'/version',
+    path: '/tidoop/v1/version',
     handler: function (request, reply) {
-        console.log("Request: GET /version");
-        var response = '{version: ' + pjson.version + '}';
+        console.log("Request: GET /tidoop/v1/version");
+        var response = '{version: ' + version + '}';
         console.log("Response: " + response);
         reply(response);
-    }
+    } // handler
+});
+
+server.route({
+    method: 'GET',
+    path: '/tidoop/v1/jobs/',
+    handler: function(request, reply) {
+        console.log('Request: GET /tidoop/v1/jobs/');
+        throw new Error('Unsupported operation');
+    } // handler
+});
+
+server.route({
+    method: 'POST',
+    path: '/tidoop/v1/jobs/',
+    handler: function (request, reply) {
+        console.log('Request: POST /tidoop/v1/jobs/ ' + JSON.stringify(request.payload));
+
+        // check the request parameters
+        // TBD
+
+        // get the jobType
+        var jobType = request.payload.job_type;
+
+        // create a jobId
+        var jobId = 'tidoop_job_' + Date.now();
+
+        // create a new job entry in the database
+        tidoopMysql.addNewJob(jobId, jobType, function(error, result) {
+            if (error) {
+                console.log('The new job could not be added to the database. Details: ' + err);
```

Comment:
You should always `reply()` to close the request flow, notifying an error or just to close the connection with the client side. If not, hapi will terminate it after some timeout for you and some kind of message would be logged :)